src/composables/useGoalData.js

```diff
@@ -0,0 +1,152 @@
+import {
+	computed,
+	inject,
+	ref,
+} from 'vue';
+
+import thankYouPageQuery from '#src/graphql/query/thankYouPage.graphql';
+import { createUserPreferences, updateUserPreferences } from '#src/util/userPreferenceUtils';
+
+import {
+	ID_BASIC_NEEDS,
+	ID_CLIMATE_ACTION,
+	ID_REFUGEE_EQUALITY,
+	ID_US_ECONOMIC_EQUALITY,
+	ID_WOMENS_EQUALITY
+} from '#src/composables/useBadgeData';
+
+const CATEGORY_TAG_MAP = {
+	[ID_BASIC_NEEDS]: ['basic_needs'],
+	[ID_CLIMATE_ACTION]: ['climate_action'],
+	[ID_REFUGEE_EQUALITY]: ['refugee_support'],
+	[ID_US_ECONOMIC_EQUALITY]: ['us_entrepreneurs'],
+	[ID_WOMENS_EQUALITY]: ['women', '#Woman-Owned Business'],
+};
+
+const GOAL_DISPLAY_MAP = {
+	[ID_BASIC_NEEDS]: 'basic needs loans',
+	[ID_CLIMATE_ACTION]: 'eco-friendly loans',
+	[ID_REFUGEE_EQUALITY]: 'refugees',
+	[ID_US_ECONOMIC_EQUALITY]: 'U.S. entrepreneurs',
+	[ID_WOMENS_EQUALITY]: 'women',
+};
+
+function getGoalDisplayName(category) {
+	return GOAL_DISPLAY_MAP[category] || 'loans';
+}
+
+function getCategoriesForTag(tag) {
+	return Object.entries(CATEGORY_TAG_MAP)
+		.filter(([, tags]) => tags.includes(tag))
+		.map(([category]) => category);
+}
+
+/**
+ * Vue composable for loading and managing user goal data
+ *
+ * @param {Array} loans - List of loans to count toward goals
+ * @returns Goal data and utilities
+ */
+export default function useGoalData(loans) {
+	const apollo = inject('apollo');
+
+	const loading = ref(true);
+	const goalState = ref({});
+	const userPreferences = ref(null);
+
+	async function loadPreferences() {
+		try {
+			const response = await apollo.query({ query: thankYouPageQuery });
+			const prefsData = response.data?.my?.userPreferences || null;
+			userPreferences.value = prefsData;
+			return prefsData ? JSON.parse(prefsData.preferences || '{}') : {};
+		} catch (error) {
+			console.error('Failed to load preferences:', error);
+			return {};
+		}
+	}
+
+	// Initialize goal state from parsed prefs (filter current year, set counts to 0)
+	function initializeGoalState(parsedPrefs) {
+		const currentYear = new Date().getFullYear();
+		const goals = parsedPrefs.goals || [];
+		goals
+			.filter(goal => new Date(goal.dateStarted).getFullYear() >= currentYear)
+			.forEach(goal => {
+				goalState.value[goal.category] = {
+					target: goal.target,
+					status: goal.status,
+					count: 0,
+					dateStarted: goal.dateStarted,
+					goalName: goal.goalName,
+					category: goal.category,
+				};
+			});
+	}
+
+	// Count loans toward goals (only current-year loans)
+	function countLoansTowardGoals() {
+		const currentYear = new Date().getFullYear();
+		loans
+			.filter(
+				loan => loan.disbursalDate
+				&& new Date(loan.disbursalDate).getFullYear() >= currentYear
+			)
+			.forEach(loan => {
+				(loan.tags || []).forEach(tag => {
+					getCategoriesForTag(tag).forEach(category => {
```

Comment:
This check actually won't work, since the goals/achievements aren't related to just tags. See the filters in `useBadgeData`. Like mentioned in Slack, you should be able to utilize achievement service endpoints, like `fetchPostCheckoutAchievements`. Not sure if all the data needed is already surfaced, but we should leverage the achievement service if we can. There is also some timing around when the achievement service updates the totals on its side, hence why we use the `fetchPostCheckoutAchievements` method, it doesn't need to wait for results to change in achievement service. We can always extend the achievement service as needed.